core/src/common/mutable_vector.rs

```diff
@@ -1,409 +0,0 @@
-// Licensed to the Apache Software Foundation (ASF) under one
-// or more contributor license agreements.  See the NOTICE file
-// distributed with this work for additional information
-// regarding copyright ownership.  The ASF licenses this file
-// to you under the Apache License, Version 2.0 (the
-// "License"); you may not use this file except in compliance
-// with the License.  You may obtain a copy of the License at
-//
-//   http://www.apache.org/licenses/LICENSE-2.0
-//
-// Unless required by applicable law or agreed to in writing,
-// software distributed under the License is distributed on an
-// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
-// KIND, either express or implied.  See the License for the
-// specific language governing permissions and limitations
-// under the License.
-
-use arrow::{
-    array::ArrayData, buffer::Buffer as ArrowBuffer, datatypes::DataType as ArrowDataType,
-};
-
-use crate::{
-    common::{bit, CometBuffer, ValueGetter, ValueSetter},
-    BinaryType, StringType, StringView, TypeTrait,
-};
-
-const DEFAULT_ARRAY_LEN: usize = 4;
-
-/// A mutable vector that can be re-used across batches.
-#[derive(Debug)]
-pub struct MutableVector {
```

Comment:
We don't need this?

Reply:
@sunchao Removed this in the original PR that I forked from.

Reply:
Yes we don't need this anymore. It was created originally to facilitate Comet specific data types, vectors and kernels (for instance leveraging array view), but we've decided to fully use DataFusion now.